hello/hello_cancellation.py

```diff
@@ -0,0 +1,84 @@
+import asyncio
+import traceback
+from datetime import timedelta
+from typing import NoReturn
+
+from temporalio import activity, workflow
+from temporalio.client import Client, WorkflowFailureError
+from temporalio.worker import Worker
+
+
+@activity.defn
+async def never_complete_activity() -> NoReturn:
+    # All long-running activities should heartbeat. Heartbeat is how
+    # cancellation is delivered from the server.
+    try:
+        while True:
+            print("Heartbeating activity")
+            activity.heartbeat()
+            await asyncio.sleep(1)
+    except asyncio.CancelledError:
+        print("Activity cancelled")
+        raise
+
+
+@activity.defn
+async def cleanup_activity() -> None:
+    print("Executing cleanup activity")
+
+
+@workflow.defn
+class CancellationWorkflow:
+    @workflow.run
+    async def run(self) -> None:
+        # Execute the forever running activity, and do a cleanup activity when
+        # it is complete (on error or cancel)
+        try:
+            await workflow.execute_activity(
+                never_complete_activity,
+                start_to_close_timeout=timedelta(seconds=1000),
+                # Always set a heartbeat timeout for long-running activities
+                heartbeat_timeout=timedelta(seconds=2),
```

Comment:
NOTE that this isn't a strict requirement, it just helps the worker configure throttling

Reply:
I'd still recommend setting it, the above comment is just a note

Reply:
Yup. I tell people to always set it as a best practice.